radicle-node/src/service.rs

```diff
@@ -571,12 +571,9 @@ where
                 if other == rid {
                     debug!(target: "service", "Ignoring redundant attempt to fetch {rid} from {from}");
                 } else {
-                    // TODO: If we can't fetch, it's because we're already fetching from
-                    // this peer. So we need to queue the request, or find another peer.
-                    error!(
-                        target: "service",
-                        "Dropping fetch for {rid} from {from}: another fetch is ongoing"
-                    );
+                    // This fetch will be queued in `Reactor.outbox`.
+                    let fetch = Message::Fetch { rid };
+                    self.reactor.write(session, fetch);
```

Comment:
What if 2+ fetch messages are queued? When the initial fetch completes, both fetch messages will be drained from the outbox, and sent to the peer. The second fetch message will be dropped, as it's received during a fetch, and the peer will be disconnected for misbehavior :thinking:

Reply:
Only one fetch message will be drained at a time, see [here](https://github.com/radicle-dev/heartwood/blob/keepsimple1/queue-fetch-342/radicle-node/src/service.rs#L1268). 

The remaining fetch messages will still be queued and drained the next time around. I am adding some test code to cover such cases.